pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -47,6 +48,20 @@ func Add(mgr manager.Manager, clusterServiceCIDR string) error {
 func newReconciler(mgr manager.Manager, clusterServiceCIDR string) (reconcile.Reconciler, error) {
 	// TODO: This should be moved out to validation for reconciler struct.
 	// 		Jira story: https://issues.redhat.com/browse/WINC-277
+	// The default client serves read requests from the cache which
+	// could be stale and result in a get call to return an older version
+	// of the object. Hence we are using a non-default-client referenced
+	// by operator-sdk.
+	cfg, err := config.GetConfig()
```

Comment:
Why can't this be mgt.GetConfig() and then build all necessary clients?

Reply:
This is what the [docs](https://sdk.operatorframework.io/docs/golang/references/client/#non-default-client) suggests to do